src/In1d.chpl

```diff
@@ -74,10 +74,11 @@ module In1d
         /* ar[{u1.size..#u2.size}] = u2; */
         ar[D.interior(u2.size)] = u2;
         // Sort unique arrays together to find duplicates
-        var order = radixSortLSD_ranks(ar);
+        // TODO could further optimize to avoid splitting into 2 arrays
```

Comment:
e.g. the following diff would allow us to avoid creating the 2 split arrays (slight faster since we avoid array creation for 2 arrays and avoid the copy):

```diff
diff --git a/src/In1d.chpl b/src/In1d.chpl
index 6df1bc6c..e448568e 100644
--- a/src/In1d.chpl
+++ b/src/In1d.chpl
@@ -74,20 +74,15 @@ module In1d
         /* ar[{u1.size..#u2.size}] = u2; */
         ar[D.interior(u2.size)] = u2;
         // Sort unique arrays together to find duplicates
-        // TODO could further optimize to avoid splitting into 2 arrays
-        var sar: [D] int;
-        var order: [D] int;
-        forall (s, o, so) in zip(sar, order, radixSortLSD(ar)) {
-          (s, o) = so;
-        }
+        var sar_order = radixSortLSD(ar);
         // Duplicates correspond to values in both arrays
         var flag: [D] bool;
-        flag[D.interior(-(D.size-1))] = (sar[D.interior(D.size-1)] == sar[D.interior(-(D.size-1))]);
+        flag[D.interior(-(D.size-1))] = (sar_order[D.interior(D.size-1)][0] == sar_order[D.interior(-(D.size-1))][0]);
         // Get the indices of values from u1 that are also in u2
         // Because sort is stable, original index of left duplicate will always be in u1
         var ret: [D] bool;
-        forall (o, f) in zip(order, flag) with (var agg = newDstAggregator(bool)) {
-            agg.copy(ret[o], f);
+        forall (o, f) in zip(sar_order, flag) with (var agg = newDstAggregator(bool)) {
+            agg.copy(ret[o[1]], f);
         }
         // Use the inverse index to map from u1 domain to ar1 domain
         var truth: [aD1] bool;

```

I think it makes the code harder to follow and no longer index-neutral, but curious for other input. Maybe from @mhmerrill in particular.

The performance difference is minor, maybe 0.5% for in1d

Reply:
@ronawho I think I'd opt for easier to understand code at this point.